packages/react/src/component-tests/IcDataEntity/IcDataEntity.cy.tsx

```diff
@@ -0,0 +1,841 @@
+  });
+
+  it("data entity visual and a11y testing", () => {
+    mount( 
+    <div style={{ margin: "16px" }}>
+    <IcDataEntity heading="Order details">
+    <IcDataRow label="Order name" value="Michael">
+      <IcLink href="#" slot="end-component">
+        Edit
+      </IcLink>
+    </IcDataRow>
+    <IcDataRow label="Drink" value="Americano">
+      <IcLink href="#" slot="end-component">
+        Edit
+      </IcLink>
+    </IcDataRow>
+    <IcDataRow label="Milk option" value="Soya milk">
+      <IcLink href="#" slot="end-component">
+        Edit
+      </IcLink>
+    </IcDataRow>
+    <IcDataRow label="Price" value="£3.25">
+      <IcLink href="#" slot="end-component">
+        Edit
+      </IcLink>
+    </IcDataRow>
+    <IcDataRow label="Payment method">
+      <IcTypography variant="body" slot="value">
+        VISA ending 5896
+      </IcTypography>
+      <IcLink href="#" slot="end-component">
+        Edit
+      </IcLink>
+    </IcDataRow>
+    <IcDataRow label="Download receipt" value="CoffeeOrder_X46w32.pdf">
+      <IcButton variant="icon" aria-label="Download" slot="end-component">
+        <SlottedSVG viewBox="0 0 24 24" 
+        height="24px" 
+        width="24px">
+          <path d="M5,20H19V18H5M19,9H15V3H9V9H5L12,16L19,9Z"
+        fill="currentColor"
+          />
+        </SlottedSVG>
+      </IcButton>
+    </IcDataRow>
+  </IcDataEntity>
+  </div>);
+    cy.checkA11yWithWait();
+    cy.compareSnapshot("alldata", DEFAULT_TEST_THRESHOLD + 0.05);
+  });
+
+  it("order name data e2e and visual testing", () => {
+    mount(
+        <div style={{ margin: "16px" }}>
+        <IcDataEntity heading="Order details">
+        <IcDataRow label="Order name" value="Michael">
+          <IcLink href="#" slot="end-component">
+            Edit 
+            </IcLink>
+         </IcDataRow>
+        </IcDataEntity>
+        </div>);
+    cy.checkHydrated("ic-data-entity");
+    cy.get('[heading="Order details"]').should(BE_VISIBLE);
+    cy.get('[label="Order name"]').should(BE_VISIBLE);
+    cy.get("ic-data-row").should(HAVE_VALUE,"Michael");
+    cy.get("ic-link").should(BE_VISIBLE).should(HAVE_TEXT,"Edit");
+    cy.compareSnapshot("nameData", DEFAULT_TEST_THRESHOLD + 0.03);
+  });
+
+  it("drink data e2e and visual testing", () => {
+    mount(
+        <div style={{ margin: "16px" }}>
+        <IcDataEntity heading="Order details">
+            <IcDataRow label="Drink" value="Americano">
+           <IcLink href="#" slot="end-component">
+        Edit
+      </IcLink>
+    </IcDataRow>
+    </IcDataEntity>
+    </div>);
+    cy.checkHydrated("ic-data-entity");
+    cy.get('[heading="Order details"]').should(BE_VISIBLE);
+    cy.get('[label="Drink"]').should(BE_VISIBLE);
+    cy.get("ic-data-row").should(HAVE_VALUE,"Americano");
+    cy.get("ic-link").should(BE_VISIBLE).should(HAVE_TEXT,"Edit");
+    cy.compareSnapshot("drinkName", DEFAULT_TEST_THRESHOLD);
+  });
+
+  it("milk data e2e and visual testing", () => {
```

Comment:
I think the it statements should be more descriptive as well e.g.
```suggestion
  it("should display with only one data row", () => {
```

Reply:
This applies to all of the if statements, not just the milk one